### DISCUSSION TITLE
Handling user logouts correctly in wave apps

### DISCUSSION THREAD

**Discussion by dulajra** [h2oai/wave#1334] [Q&A] [answered]
I have a Wave app with OIDC auth enabled. Wave app uses `q.auth.access_token` to authenticate with the MLOps backend. All is working fine in the usual flow. 

I have registered `system.logout` event as mentioned in the [docs](https://wave.h2o.ai/docs/routing#handling-user-logout). But if I log in to keycloak and invalidate the session for the same user the above event is not triggered. Isn't this scenario supported by `system.logout` event? If it's not then is there any other method provided by the SDK to handle such a scenario? 

cc: @lo5

**Comment by azim-b** [comment]
https://github.com/h2oai/wave/discussions/1325#discussioncomment-2509569

**Reply by dulajra** [reply]
@azim-b The above thread is on how to log out from a wave app. But my question is different. I want to know if the user session has been already invalidated somehow is there a way to notify it to the app so that the app can redirect the user to the sign-in page for a re-login.

**Comment by lo5** [comment, marked as answer]
> Isn't this scenario supported by system.logout event?

No. 

That would have to be a new feature in Wave.

**Reply by dulajra** [reply]
@lo5 Is there any plan to implement such a feature? Shall I open a feature request on GitHub?
